tests/unit/store/createStore.ts

```diff
@@ -370,5 +380,80 @@ registerSuite({
 				assert.deepEqual(result, ids, 'Should have returned all deleted ids');
 			}).then(dfd.resolve);
 		}
+	},
+
+	'async storage': {
+		'async operation should not be done immediately.'(this: any) {
+			const{ dfd, store } = getStoreWithAsyncStorage(this, { put: 21 });
+
+			const start = Date.now();
+			store.add(createData()).then(dfd.callback(function(result: ItemType[]) {
+				const finish = Date.now();
+				assert.isAbove(finish - start, 19 );
+				dfd.resolve();
+			}));
+		},
+		'should complete initial add before subsequent operations'(this: any) {
+			const dfd = this.async(1000);
+			const asyncStorage = createAsyncStorage();
+			const store = createStore({
+				storage: asyncStorage,
+				data: createData()
+			});
+
+			store.get(['1', '2', '3']).then(dfd.callback(function(items: ItemType[]) {
+				assert.deepEqual(items, createData(), 'Didn\'t retrieve items from async add');
+			}));
+		},
+		'failed initial add should not prevent subsequent operations'(this: any) {
+			const dfd = this.async(1000);
+			let fail = true;
+			const asyncStorage = createAsyncStorage
+				.around('add', function(add: () => Promise<ItemType>) {
+					return function(this: any) {
+						if (fail) {
+							fail = false;
+							return Promise.reject(Error('Error'));
+						}
+						else {
+							return add.apply(this, arguments);
+						}
+					};
+				})();
+			const data = createData();
+			const store = createStore({
+				storage: asyncStorage,
+				data: data
+			});
+
+			store.add(data).then(function() {
+				store.get(['1', '2', '3']).then(dfd.callback(function(items: ItemType[]) {
+					assert.isFalse(fail, 'Didn\'t fail for first operation');
+					assert.deepEqual(items, data, 'Didn\'t retrieve items from add following failed initial add');
+				}));
+			});
+		},
+		'fetch should not return items when it is done before add.'(this: any) {
+			const { dfd, store } = getStoreWithAsyncStorage(this, { put: 20, fetch: 10 });
+			store.add(createData());
+			store.fetch().then(function(storeData) {
+				assert.lengthOf(storeData, 0, 'should not have retrieved items');
+			}).then(dfd.resolve);
+		},
+		'async operations should be done in the order specified by the user.'(this: any) {
+			const{ dfd, store } = getStoreWithAsyncStorage(this);
+
+			store.add(createData()).then(function(result) {
+				assert.deepEqual(result, createData(), 'Should have returned all added items');
+			}).then(function() {
+				return store.put(createUpdates()[0]);
```

Comment:
Because the store operations could change the cache. `createUpdate()` allows us to always have the original expected test data.